lib/ansible/modules/cloud/azure/azure_rm_webapp.py

```diff
@@ -0,0 +1,971 @@
+            - Unique name of the app to create or update. To create or update a deployment slot, use the {slot} parameter.
+        required: True
+
+    location:
+        description:
+            - Resource location. If not set, location from the resource group will be used as default.
+
+    plan:
+        description:
+            - App service plan. Required for creation.
+            - It can be name of existing app service plan in same resource group as web app.
+            - "It can be resource id of existing app service plan. eg.,
+              /subscriptions/<subs_id>/resourceGroups/<resource_group>/providers/Microsoft.Web/serverFarms/<plan_name>"
+            - It can be a dict which contains C(name), C(resource_group), C(sku), C(is_linux) and C(number_of_workers).
+              name. Name of app service plan.
+              resource_group. Resource group name of app service plan.
+              sku. SKU of app service plan. For allowed sku, please refer to https://azure.microsoft.com/en-us/pricing/details/app-service/linux/.
+              is_linux. Indicate is linux app service plan. type bool. default False.
+              number_of_workers. Number of workers.
+
+    frameworks:
+        description:
+            - Set of run time framework settings. Each setting is a dictionary.
+            - See https://docs.microsoft.com/en-us/azure/app-service/app-service-web-overview for more info.
+        suboptions:
+            name:
+                description:
+                    - Name of the framework.
+                    - Supported framework list for Windows web app and Linux web app is different.
+                    - For Windows web app, supported names(June 2018) java, net_framework, php, python, node. Multiple framework can be set at same time.
+                    - For Linux web app, supported names(June 2018) java, ruby, php, dotnetcore, node. Only one framework can be set.
+                    - Java framework is mutually exclusive with others.
+                choices:
+                    - java
+                    - net_framework
+                    - php
+                    - python
+                    - ruby
+                    - dotnetcore
+                    - node
+            version:
+                description:
+                    - Version of the framework. For linux web app supported value, see https://aka.ms/linux-stacks for more info.
+                    - net_framework supported value sample, 'v4.0' for .NET 4.6 and 'v3.0' for .NET 3.5.
+                    - php supported value sample, 5.5, 5.6, 7.0.
+                    - python supported value sample, e.g., 5.5, 5.6, 7.0.
+                    - node supported value sample, 6.6, 6.9.
+                    - dotnetcore supported value sample, 1.0, 1,1, 1.2.
+                    - ruby supported value sample, 2.3.
+            settings:
+                description:
+                    - List of settings of the framework.
```

Comment:
Should add a comment saying this is only supported by specific frameworks like C(java)